meta-oe/recipes-devtools/php/php_8.0.12.bb

```diff
@@ -259,7 +259,7 @@ do_install:append:class-native() {
     create_wrapper ${D}${bindir}/php \
         PHP_PEAR_SYSCONF_DIR=${sysconfdir}/
 
-    if [ "$MACHINE_ARCH" == "x86" || "$MACHINE_ARCH" == "x86-64" ]; then
+    if [ "$MACHINE_ARCH" == "x86" ] || [ "$MACHINE_ARCH" == "x86-64" ]; then
```

Comment:
`if [ "$MACHINE_ARCH" == "x86" ] || [ "$MACHINE_ARCH" == "x86-64" ]; then`
works with dash as well, right?

And where this MACHINE_ARCH variable is being set? if it's MACHINE_ARCH bitbake variable then it should be inside ${} and it doesn't make much sense to me as it's inside:
`do_install:append:class-native()` and _native_ cannot be MACHINE specific.

Reply:
yes, when I looked a bit deeper, I realised this too. I am testing this in a CI right now

https://github.com/YoeDistro/meta-openembedded/commit/58231333a0dd6d4464b06314484e778578802eb4

Reply:
Yes, I will admit I am out of my element here with trying to get this working across multiple platforms.  I know it works when compiling on X86-64 for an X86-64 target, but getting it working for other host architectures is much more involved.

Reply:
@kraj The `minilua` binary is only needed by the target during compilation, it does not need to be added to the resulting image.  The patch `0011-opcache-jit-use-minilua-in-sysroot.patch` tells the target to just use the already-compiled `minilua` binary instead of trying to build it.  The native recipe is building `minilua`.

Is there a way to build the `minilua` for the target arch and have the Makefile run this binary under QEMU?

Reply:
yes please see meta-oe/recipes-support/espeak/espeak_1.48.04.bb